cnap/core/filedb.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module provides an object-oriented design for file database to load video file.
+It implements an abstract base class `FileDatabase` and a concrete filedatabase class
+`LocalFileDatabase`.
+
+`FileDatabase` serves as a blueprint for custom filedatabase implementations, while
+`LocalFileDatabase` provide an implementation for local filedatabase to get video 
+file locally.
+
+These classes can be easily extended or modified to accommodate new filedatabase type.
+"""
+
+import os
+import logging
+from abc import ABC, abstractmethod
+
+LOG = logging.getLogger(__name__)
+CURR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+class FileDatabase(ABC):
+    """
+    Abstract base class for creating custom filedatabase implementations.
+    """
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def get_file(self, filename: str) -> str:
+        """
+        Get/Download file from file database and return the local path for file.
+
+        Args:
+            filename: The name of file to get/Download.
+
+        Returns:
+            str: The local path for file.
+        """
+        raise NotImplementedError("Subclasses should implement get_file() method.")
+
+class LocalFileDatabase(FileDatabase):
+    """
+    Local filedata base implementation for getting video file locally.
+    """
+
+    def get_file(self, filename: str) -> str:
+        """
+        Get the local path for file.
+
+        Args:
+            filename: The name of file to get.
+
+        Returns:
+            str: The local path for file.
+            None: Invalid root directory or failed to find the given file.
+        """
+        root_dir = os.path.abspath(os.path.join(CURR_DIR, "../../demo/video"))
+        if not os.path.exists(root_dir):
+            LOG.error("Invalid root directory for local file database.")
+            return None
```

Comment:
Raise the error may be better than return no-meaning none.

Reply:
Thanks, fixed.